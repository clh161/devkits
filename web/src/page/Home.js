// @flow strict
import React from 'react';
import type { Node } from 'react';
import { Grid, Link, Typography } from '@material-ui/core';
import { Helmet } from 'react-helmet';
export default function Home(): Node {
  return (
    <div>
      <Helmet>
        <title>Devkits</title>
      </Helmet>
      <Grid container spacing={3}>
        <Grid item xs={12}>
          <Typography block component="h1" variant="h4">
            Devkits
          </Typography>
        </Grid>
        <Grid item xs={12}>
          <Typography variant="body1">
            An opensource collection of development tools
          </Typography>
        </Grid>
        <Grid item xs={12}>
          <Typography variant="link">
            <Link href="https://github.com/clh161/devkits" target="_blank">
              Github Repository
            </Link>
          </Typography>
        </Grid>
      </Grid>
    </div>
  );
}
