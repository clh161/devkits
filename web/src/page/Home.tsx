import { Grid, Link, Typography } from '@material-ui/core';
import React, { ReactElement } from 'react';
export default function Home(): ReactElement {
  return (
    <Grid container spacing={3}>
      <Grid item xs={12}>
        <Typography variant="body1">
          An opensource project of collection of development tools
        </Typography>
      </Grid>
      <Grid item xs={12}>
        <Typography variant="overline">
          <Link href="https://github.com/clh161/devkits" target="_blank">
            Github Repository
          </Link>
        </Typography>
      </Grid>
    </Grid>
  );
}
