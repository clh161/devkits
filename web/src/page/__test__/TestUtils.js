// @flow strict

export function mockLocation(mockPath: string) {
  jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: () => ({
      pathname: mockPath,
    }),
  }));
}
