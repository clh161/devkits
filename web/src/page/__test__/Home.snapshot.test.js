// @flow

import React from 'react';
import { Helmet } from 'react-helmet';
import renderer from 'react-test-renderer';

import Home from '../Home';
import { mockLocation } from './TestUtils';
mockLocation('/');
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/',
  }),
}));
it('renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(Helmet.peek()).toMatchSnapshot();
});
