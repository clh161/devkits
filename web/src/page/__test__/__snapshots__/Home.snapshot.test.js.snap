// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h1
        block={true}
        className="MuiTypography-root MuiTypography-h4"
      >
        Devkits
      </h1>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        An opensource project of collection of development tools
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <span
        className="MuiTypography-root MuiTypography-overline"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://github.com/clh161/devkits"
          onBlur={[Function]}
          onFocus={[Function]}
          target="_blank"
        >
          Github Repository
        </a>
      </span>
    </div>
  </div>
</div>
`;
